From 2b4343838dbc36552b7f7f18ee064dc9c867564a Mon Sep 17 00:00:00 2001
From: Dinesh Kumar Mirche <dmirche@habana.ai>
Date: Mon, 11 Sep 2023 15:30:45 +0300
Subject: [PATCH] [SW-66359] preboot: fix compilation errors

This patch fixes the compilations errors in gaudi2 security
files by adding appropriate header files. These compilation
issues are because some security structures and enum declarations
are moved to common header file from platform specific

Change-Id: If46888badba0ed6b6f2e97746549ef600ddc44c3
Signed-off-by: Dinesh Kumar Mirche <dmirche@habana.ai>
---

diff --git a/app/ext_rom/erom_tpm.c b/app/ext_rom/erom_tpm.c
index 5193f00..9127e9d 100644
--- a/app/ext_rom/erom_tpm.c
+++ b/app/ext_rom/erom_tpm.c
@@ -24,6 +24,7 @@
 #include <gaudi2/arc/dev_security.h>
 #include <sha512.h>
 #include <string.h>
+#include <hl_sec_if.h>
 
 static void erom_get_ppboot_data(uint8_t **offset)
 {
diff --git a/arch/gaudi2/arc/common.c b/arch/gaudi2/arc/common.c
index e2c7f9e..7260092 100644
--- a/arch/gaudi2/arc/common.c
+++ b/arch/gaudi2/arc/common.c
@@ -55,6 +55,7 @@
 #include <gaudi2_bootrom.h>
 #include <hl_boot_desc.h>
 #include <boot_info.h>
+#include <hl_sec_if.h>
 
 #define PRODUCT_ID_STR_SIZE			0x8
 
diff --git a/arch/gaudi2/common/security.c b/arch/gaudi2/common/security.c
index 2826594..566f480 100644
--- a/arch/gaudi2/common/security.c
+++ b/arch/gaudi2/common/security.c
@@ -57,6 +57,8 @@
 #include <dcore0_hmmu0_mmu_masks.h>
 #include <pll.h>
 #include <gaudi2_fw_if.h>
+#include <hl_sec_if.h>
+
 #include "common.h"
 
 static struct hl_sec_rrs_cfg hl_sec_lbw_if_rtr_rrs = {
